### PR TITLE
Remove schema files, redirect Makefile to new repo

### PR DESCRIPTION
**Pull request by clavin-xlnx**
Signed-off-by: Chris Lavin <clavin@xilinx.com>

### CODE REVIEW

**Comment by clavin-xlnx** [comment]
@litghost - Would you mind reviewing the change to redirect RapidWright to point to SymbiFlow/fpga-interchange-schema?

File: interchange/Makefile [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
 	wget https://search.maven.org/remotecontent?filepath=org/capnproto/runtime/0.1.4/runtime-0.1.4.jar -O $(JAVA_CP_RUNTIME)
 
 $(JAVA_SCHEMA):
 	mkdir -p $(JAVA_SCHEMA_DIR)
 	wget https://raw.githubusercontent.com/capnproto/capnproto-java/master/compiler/src/main/schema/capnp/java.capnp -O $(JAVA_SCHEMA)
+
+$(INTERCHANGE_REPO_PATH):
+	cd $(INTERCHANGE_REPO_PATH)/.. && git clone $(INTERCHANGE_CLONE_PATH)
+
+update_interchange: $(INTERCHANGE_REPO_PATH)
```
Comment (litghost) [inline interchange/Makefile:27]: This should be:
`update_interchange: | $(INTERCHANGE_REPO_PATH)`

File: interchange/Makefile [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
-JAVA_SCHEMA_DIR=schema/capnp
-JAVA_SCHEMA=$(JAVA_SCHEMA_DIR)/java.capnp
+SRCS_WITH_PATH = $(addprefix $(INTERCHANGE_REPO_PATH)/$(INTERCHANGE_NAME)/, $(SRCS))
+DST = ../src/com/xilinx/rapidwright/$(INTERCHANGE_NAME)
+JAVA_CP_RUNTIME = ../jars/runtime-0.1.4.jar
+JAVA_SCHEMA_DIR = schema/capnp
+JAVA_SCHEMA = $(JAVA_SCHEMA_DIR)/java.capnp
 
-all: $(JAVA_CP_RUNTIME) $(JAVA_SCHEMA)
+all: $(JAVA_CP_RUNTIME) $(JAVA_SCHEMA) $(INTERCHANGE_REPO_PATH)
```
Comment (litghost) [inline interchange/Makefile:13]: `all: $(JAVA_CP_RUNTIME) $(JAVA_SCHEMA) | $(INTERCHANGE_REPO_PATH)`

File: interchange/Makefile (hunk 1 above)
Comment (clavin-xlnx) [reply interchange/Makefile]: Thanks!  Updated.

**Comment by litghost** [comment]
In a clean checkout, I get the following error:
```
$ make
mkdir -p ../jars
wget https://search.maven.org/remotecontent?filepath=org/capnproto/runtime/0.1.4/runtime-0.1.4.jar -O ../jars/runtime-0.1.4.jar
--2021-02-18 13:45:07--  https://search.maven.org/remotecontent?filepath=org/capnproto/runtime/0.1.4/runtime-0.1.4.jar
Resolving search.maven.org (search.maven.org)... 18.208.65.80, 52.203.117.103
Connecting to search.maven.org (search.maven.org)|18.208.65.80|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://repo1.maven.org:443/fromsearch?filepath=org/capnproto/runtime/0.1.4/runtime-0.1.4.jar [following]
--2021-02-18 13:45:07--  https://repo1.maven.org/fromsearch?filepath=org/capnproto/runtime/0.1.4/runtime-0.1.4.jar
Resolving repo1.maven.org (repo1.maven.org)... 199.232.192.209, 199.232.196.209
Connecting to repo1.maven.org (repo1.maven.org)|199.232.192.209|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://repo1.maven.org/maven2/org/capnproto/runtime/0.1.4/runtime-0.1.4.jar [following]
--2021-02-18 13:45:07--  https://repo1.maven.org/maven2/org/capnproto/runtime/0.1.4/runtime-0.1.4.jar
Reusing existing connection to repo1.maven.org:443.
HTTP request sent, awaiting response... 200 OK
Length: 104179 (102K) [application/java-archive]
Saving to: ‘../jars/runtime-0.1.4.jar’

../jars/runtime-0.1.4.jar                                              100%[==========================================================================================================================================================================>] 101.74K  --.-KB/s    in 0.003s  

2021-02-18 13:45:07 (37.6 MB/s) - ‘../jars/runtime-0.1.4.jar’ saved [104179/104179]

mkdir -p schema/capnp
wget https://raw.githubusercontent.com/capnproto/capnproto-java/master/compiler/src/main/schema/capnp/java.capnp -O schema/capnp/java.capnp
--2021-02-18 13:45:07--  https://raw.githubusercontent.com/capnproto/capnproto-java/master/compiler/src/main/schema/capnp/java.capnp
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1459 (1.4K) [text/plain]
Saving to: ‘schema/capnp/java.capnp’

schema/capnp/java.capnp                                                100%[==========================================================================================================================================================================>]   1.42K  --.-KB/s    in 0s      

2021-02-18 13:45:08 (12.4 MB/s) - ‘schema/capnp/java.capnp’ saved [1459/1459]

cd ~/cat_x/clean/RapidWright/interchange/fpga-interchange-schema/.. && git clone https://github.com/SymbiFlow/fpga-interchange-schema.git
/bin/sh: line 1: cd: ~/cat_x/clean/RapidWright/interchange/fpga-interchange-schema/..: No such file or directory
make: *** [Makefile:26: ~/cat_x/clean/RapidWright/interchange/fpga-interchange-schema] Error 1
```

**Comment by litghost** [comment]
I believe this is the fix:
```
diff --git a/interchange/Makefile b/interchange/Makefile
index 17be107..ed0f268 100644
--- a/interchange/Makefile
+++ b/interchange/Makefile
@@ -1,7 +1,7 @@
 CAPNPC = capnpc
 INTERCHANGE_NAME = interchange
 INTERCHANGE_REPO_NAME = fpga-$(INTERCHANGE_NAME)-schema
-INTERCHANGE_REPO_PATH = $(shell pwd)/$(INTERCHANGE_REPO_NAME)
+INTERCHANGE_REPO_PATH = $(realpath .)/$(INTERCHANGE_REPO_NAME)
 INTERCHANGE_CLONE_PATH = https://github.com/SymbiFlow/$(INTERCHANGE_REPO_NAME).git
 SRCS = LogicalNetlist.capnp DeviceResources.capnp PhysicalNetlist.capnp References.capnp
 SRCS_WITH_PATH = $(addprefix $(INTERCHANGE_REPO_PATH)/$(INTERCHANGE_NAME)/, $(SRCS))
@@ -23,7 +23,7 @@ $(JAVA_SCHEMA):
        wget https://raw.githubusercontent.com/capnproto/capnproto-java/master/compiler/src/main/schema/capnp/java.capnp -O $(JAVA_SCHEMA)
 
 $(INTERCHANGE_REPO_PATH):
-       cd $(INTERCHANGE_REPO_PATH)/.. && git clone $(INTERCHANGE_CLONE_PATH)
+       cd $(dir $(INTERCHANGE_REPO_PATH)) && git clone $(INTERCHANGE_CLONE_PATH)
 
 update_interchange: | $(INTERCHANGE_REPO_PATH)
        cd $(INTERCHANGE_REPO_PATH) && git pull
```

**Comment by litghost** [comment]
There are other bugs, fixes:
```
diff --git a/interchange/Makefile b/interchange/Makefile
index 17be107..0881d06 100644
--- a/interchange/Makefile
+++ b/interchange/Makefile
@@ -1,18 +1,17 @@
 CAPNPC = capnpc
 INTERCHANGE_NAME = interchange
 INTERCHANGE_REPO_NAME = fpga-$(INTERCHANGE_NAME)-schema
-INTERCHANGE_REPO_PATH = $(shell pwd)/$(INTERCHANGE_REPO_NAME)
+INTERCHANGE_REPO_PATH = $(realpath .)/$(INTERCHANGE_REPO_NAME)
 INTERCHANGE_CLONE_PATH = https://github.com/SymbiFlow/$(INTERCHANGE_REPO_NAME).git
 SRCS = LogicalNetlist.capnp DeviceResources.capnp PhysicalNetlist.capnp References.capnp
-SRCS_WITH_PATH = $(addprefix $(INTERCHANGE_REPO_PATH)/$(INTERCHANGE_NAME)/, $(SRCS))
-DST = ../src/com/xilinx/rapidwright/$(INTERCHANGE_NAME)
+DST = $(realpath ..)/src/com/xilinx/rapidwright/$(INTERCHANGE_NAME)
 JAVA_CP_RUNTIME = ../jars/runtime-0.1.4.jar
 JAVA_SCHEMA_DIR = schema/capnp
 JAVA_SCHEMA = $(JAVA_SCHEMA_DIR)/java.capnp
 
 all: $(JAVA_CP_RUNTIME) $(JAVA_SCHEMA) | $(INTERCHANGE_REPO_PATH)
        mkdir -p $(DST)
-       $(CAPNPC) -ojava:$(DST) $(SRCS_WITH_PATH) -I schema
+       cd $(INTERCHANGE_REPO_PATH)/$(INTERCHANGE_NAME) && $(CAPNPC) -ojava:$(DST) $(SRCS) -I $(realpath .)/schema
 
 $(JAVA_CP_RUNTIME):
        mkdir -p ../jars
@@ -23,7 +22,7 @@ $(JAVA_SCHEMA):
        wget https://raw.githubusercontent.com/capnproto/capnproto-java/master/compiler/src/main/schema/capnp/java.capnp -O $(JAVA_SCHEMA)
 
 $(INTERCHANGE_REPO_PATH):
-       cd $(INTERCHANGE_REPO_PATH)/.. && git clone $(INTERCHANGE_CLONE_PATH)
+       cd $(dir $(INTERCHANGE_REPO_PATH)) && git clone $(INTERCHANGE_CLONE_PATH)^M
 
 update_interchange: | $(INTERCHANGE_REPO_PATH)
        cd $(INTERCHANGE_REPO_PATH) && git pull
```

**Comment by clavin-xlnx** [comment]
Thanks, after the jar dir update, I realized there were other issues.  I'm incorporating your suggestions now.

**Comment by clavin-xlnx** [comment]
Ok, sorry for the bumpy submissions, should be good now.

**Review by litghost — APPROVED** [review]
LGTM

**Pull request merged**